build.zig

```diff
@@ -0,0 +1,175 @@
+// THIS MATERIAL IS PROVIDED AS IS, WITH ABSOLUTELY NO WARRANTY EXPRESSED
+// OR IMPLIED.  ANY USE IS AT YOUR OWN RISK.
+//
+// Permission is hereby granted to use or copy this program
+// for any purpose, provided the above notices are retained on all copies.
+// Permission to modify the code and to distribute modified code is granted,
+// provided the above notices are retained, and a notice that the code was
+// modified is included with the above copyright notice.
+
+const std = @import("std");
+const print = @import("std").debug.print;
+
+pub fn build(b: *std.build.Builder) void {
+    const optimize = b.standardOptimizeOption(.{});
+    const target = b.standardTargetOptions(.{});
+    const t = target.toTarget();
+    var default_enable_threads = true;
+    if (t.isWasm()) { // matches both emscripten and wasi
+        default_enable_threads = false;
+    }
```

Comment:
We need an option to choose between static and shared library builds. Default is the latter one.
E.g. BUILD_SHARED_LIBS (or enable_shared).
It is fine for this PR to fail build if shared lib build is unsupported for now.

Reply:
I have added in support for shared libs now!